pkg/apis/crd/v1alpha1/types.go

```diff
@@ -354,3 +355,146 @@ type BGPPeer struct {
+type PacketCaptureFirstNConfig struct {
+	Number int32 `json:"number"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type PacketCaptureList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []PacketCapture `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type PacketCapture struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              PacketCaptureSpec   `json:"spec"`
+	Status            PacketCaptureStatus `json:"status"`
+}
+
+type CaptureConfig struct {
+	// FirstN means we only capture first N packets from the target traffic.
+	// At the moment this is the only supported configuration and it is required for every capture.
+	FirstN *PacketCaptureFirstNConfig `json:"firstN,omitempty"`
+}
+
+// PacketCaptureFileServer specifies the PacketCapture file server information.
+type PacketCaptureFileServer struct {
+	// The URL of the file server. It is set with format: scheme://host[:port][/path],
+	// e.g., https://api.example.com:8443/v1/packets/. Currently only `sftp` protocol is supported.
+	URL string `json:"url"`
+}
+
+type PacketCaptureSpec struct {
+	// Timeout is the timeout for this capture session. If not specified, defaults to 60s.
+	Timeout       *uint16       `json:"timeout,omitempty"`
+	CaptureConfig CaptureConfig `json:"captureConfig"`
+	// Source is the traffic source we want to perform capture on. Both `Source` and `Destination` is required
+	// for a capture session, and at least one `Pod` should be present either in the source or the destination.
+	Source      Source      `json:"source"`
+	Destination Destination `json:"destination"`
+	// Packet defines what kind of traffic we want to capture between the source and destination. If not specified,
+	// all kinds of traffic will count.
+	Packet *Packet `json:"packet,omitempty"`
+	// FileServer specifies the sftp url config for a file server. If present, captured packets will be uploaded to this server.
+	// If not, the packet capture results will only be present as a file in the antrea-agent container.
+	// When the capture finished, the path information will be shown in `.status.PacketsFilePath`.
+	FileServer *PacketCaptureFileServer `json:"fileServer,omitempty"`
+}
+
+type PacketCaptureStatus struct {
+	// NumberCaptured records how many packets have been captured. If it reaches the target number, the capture
+	// can be considered as finished.
+	NumberCaptured int32 `json:"numberCaptured"`
+	// FilePath is the file path where the captured packets are stored. The format is: "<antrea-agent-pod-name>:<path>".
+	// If `.spec.FileServer` is present, this file will also be uploaded to the target location. This file
+	// will be removed after the PacketCapture CR is deleted.
```

Comment:
Consider this explanation:

FilePath specifies the location where captured packets are stored. It can either be a URL to download the pcap file (if "Spec.FileServer" is specified) or a local file path on the antrea-agent Pod where the packet was captured, formatted as <antrea-agent-pod-name>:<path>. When using a local file path, the file will be automatically removed after the PacketCapture resource is deleted.